crates/rover-client/src/shared/check_response.rs

```diff
@@ -1,159 +1,336 @@
+use console::style;
 use std::fmt::{self, Display};
 use std::str::FromStr;
 
-use crate::shared::GraphRef;
+use crate::shared::Diagnostic;
 use crate::RoverClientError;
 
+use rover_std::Style;
+
 use prettytable::format::consts::FORMAT_BOX_CHARS;
 use serde::{Deserialize, Serialize};
 
 use prettytable::{row, Table};
 use serde_json::{json, Value};
 
 #[derive(Debug, Serialize, Clone, Eq, PartialEq)]
-pub enum CheckResponse {
-    OperationCheckResponse(OperationCheckResponse),
-    SkipOperationsCheckResponse(SkipOperationsCheckResponse),
+pub struct CheckWorkflowResponse {
+    pub default_target_url: String,
+    pub maybe_core_schema_modified: Option<bool>,
```

Comment:
So for `rover graph check` (ie. non-federated checks) this value is empty in the check response but is explicitly set to false ([here](https://github.com/apollographql/rover/blob/main/crates/rover-client/src/operations/graph/check_workflow/runner.rs#L104-L109)). I figured since we reserve the term `core` schema to mean the "built" schema it didn't feel right having this be `false` when it should be `None`? ie. the absence of a `core` schema.

I can make it non-null or maybe rename it. But what i had in mind is `Some(true)` means the built schema changed in some way, `Some(false)` means the built schema did not change, and `None` means there was no schema to build which is the case for `rover graph check`.

Reply:
that checks out - would you mind adding a comment explaining this?